src/Xtools/RFA.php

```diff
@@ -0,0 +1,203 @@
+     * @param array  $matches Pointer to an array where we stash results
+     *
+     * @TODO: Make this cleaner
+     *
+     * @return int
+     */
+    protected function findSig($input, &$matches)
+    {
+        //Supports User: and User talk: wikilinks, {{fullurl}},
+        // unsubsted {{unsigned}}, unsubsted {{unsigned2}},
+        // anything that looks like a custom sig template
+        // TODO: Cross-wiki this sucker
+        $regexp
+            = //1: Normal [[User:XX]] and [[User talk:XX]]
+            "/\[\[[Uu]ser(?:[\s_][Tt]alk)?\:([^\]\|\/]*)(?:\|[^\]]*)?\]\]"
+            //2: {{fullurl}} and {{unsigned}} templates
+            . "|\{\{(?:[Ff]ullurl\:[Uu]ser(?:[\s_][Tt]alk)?\:|"
+            . "[Uu]nsigned\|)([^\}\|]*)(?:|[\|\}]*)?\}\}"
+            //3: {{User:XX/sig}} templates
+            . "|(?:\{\{)[Uu]ser(?:[\s_][Tt]alk)?\:([^\}\/\|]*)"
+            //4: {{unsigned2|Date|XX}} templates
+            . "|\{\{[Uu]nsigned2\|[^\|]*\|([^\}]*)\}\}"
+            //5: [[User:XX/sig]] links (compromise measure)
+            . "|(?:\[\[)[Uu]ser\:([^\]\/\|]*)\/[Ss]ig[\|\]]/";
+
+        return preg_match_all(
+            $regexp,
+            $input,
+            $matches,
+            PREG_OFFSET_CAPTURE
+        );
+    }
+
+    /**
+     * RFA constructor.
+     *
+     * @param string      $rawWikiText      The text of the page we're parsing
+     * @param array       $section_array    Section names that we're looking for
+     * @param string      $user_namespace   Plain text of the user namespace
+     * @param string      $date_regexp      Valid Regular Expression for the end date
+     * @param string|null $user_looking_for User we're trying to find.
+     */
+    public function __construct(
+        $rawWikiText,
+        $section_array = ["Support", "Oppose", "Neutral"],
+        $user_namespace = "User",
+        $date_regexp = "final .*end(?:ing|ed)?(?: no earlier than)? (.*?)? \(UTC\)",
+        $user_looking_for = null
+    ) {
+        $this->user_looking_for = $user_looking_for;
+
+        $lines = explode("\n", $rawWikiText);
+
+        $keys = join("|", $section_array);
+
+        $lastSection = "";
+
+        foreach ($lines as $line) {
+            if (preg_match("/={1,6}\s?($keys)\s?={1,6}/i", $line, $matches)) {
+                $lastSection = strtolower($matches[1]);
+            } elseif ($lastSection == ""
+                && preg_match(
+                    "/$date_regexp/",
+                    $line,
+                    $matches
+                )
+            ) {
+                $this->endDate = $matches[1];
+            } elseif ($lastSection != ""
+                && preg_match("/^\s*#?:.*/i", $line) === 0
+            ) {
+                $this->findSig($line, $matches);
+                if (!isset($matches[1][0])) {
+                    continue;
+                }
+                $foundUser = trim($matches[1][0][0]);
+                $this->data[$lastSection][] = $foundUser;
```

Comment:
`$this->data` is never initialized, so if there are no results there's a `Invalid argument supplied for foreach()` down on line 142.

And also: all this in the constructor? Personally I'd move it to another method that's called when required. But whichever; it works this way too! :-)